github/resource_github_repository_pull_request.go

```diff
@@ -0,0 +1,294 @@
+	if err != nil {
+		return err
+	}
+
+	pullRequest, _, err := client.PullRequests.Get(ctx, owner, repository, number)
+	if err != nil {
+		return err
+	}
+
+	d.Set("number", pullRequest.GetNumber())
+
+	if head := pullRequest.GetHead(); head != nil {
+		d.Set("head_ref", head.GetRef())
+		d.Set("head_sha", head.GetSHA())
+	} else {
+		// Totally unexpected condition. Better do that than segfault, I guess?
+		log.Printf("[WARN] Head branch missing, expected %s", d.Get("head_ref"))
+		d.SetId("")
+		return nil
+	}
+
+	if base := pullRequest.GetBase(); base != nil {
+		d.Set("base_ref", base.GetRef())
+		d.Set("base_sha", base.GetSHA())
+	} else {
+		// Seme logic as with the missing head branch.
+		log.Printf("[WARN] Base branch missing, expected %s", d.Get("base_ref"))
+		d.SetId("")
+		return nil
+	}
+
+	d.Set("body", pullRequest.GetBody())
+	d.Set("title", pullRequest.GetTitle())
+	d.Set("draft", pullRequest.GetDraft())
+	d.Set("maintainer_can_modify", pullRequest.GetMaintainerCanModify())
+	d.Set("number", pullRequest.GetNumber())
+	d.Set("state", pullRequest.GetState())
+	d.Set("opened_at", pullRequest.GetCreatedAt().Unix())
+	d.Set("updated_at", pullRequest.GetUpdatedAt().Unix())
+
+	if user := pullRequest.GetUser(); user != nil {
+		d.Set("opened_by", user.GetLogin())
+	}
+
+	labels := []string{}
+	for _, label := range pullRequest.Labels {
+		labels = append(labels, label.GetName())
+	}
+	d.Set("labels", labels)
+
+	return nil
+}
+
+func resourceGithubRepositoryPullRequestUpdate(d *schema.ResourceData, meta interface{}) error {
+	ctx := context.TODO()
+	client := meta.(*Owner).v3client
+
+	owner, repository, number, err := parsePullRequestID(d)
+	if err != nil {
+		return err
+	}
+
+	update := &github.PullRequest{
+		Title:               github.String(d.Get("title").(string)),
+		Body:                github.String(d.Get("body").(string)),
+		MaintainerCanModify: github.Bool(d.Get("maintainer_can_modify").(bool)),
+	}
+
+	if d.HasChange("base_ref") {
+		update.Base = &github.PullRequestBranch{
+			Ref: github.String(d.Get("base_ref").(string)),
+		}
+	}
+
+	_, _, err = client.PullRequests.Edit(ctx, owner, repository, number, update)
+	if err == nil {
+		return resourceGithubRepositoryPullRequestRead(d, meta)
+	}
+
+	errors := []string{fmt.Sprintf("could not update the Pull Request: %v", err)}
+
+	if err := resourceGithubRepositoryPullRequestRead(d, meta); err != nil {
+		errors = append(errors, fmt.Sprintf("could not read the Pull Request after the failed update: %v", err))
+	}
+
+	return fmt.Errorf(strings.Join(errors, ", "))
+}
+
+func resourceGithubRepositoryPullRequestDelete(d *schema.ResourceData, meta interface{}) error {
+	// It's not entirely clear how to treat PR deletion according to Terraform's
+	// CRUD semantics. The approach we're taking here is to close the PR unless
+	// it's already closed or merged. Merging it feels intuitively wrong in what
+	// effectively is a destructor.
```

Comment:
👍🏾  lets not merge by default.  Perhaps we can provide an option to do so in the future if this becomes a common ask.